lockfile/lockfile-utils/src/pkgSnapshotToResolution.ts

```diff
@@ -13,7 +14,7 @@ export function pkgSnapshotToResolution (
 ): Resolution {
   if (
     Boolean((pkgSnapshot.resolution as TarballResolution).type) ||
-    (pkgSnapshot.resolution as TarballResolution).tarball?.startsWith('file:')
+    (pkgSnapshot.resolution as TarballResolution).tarball?.startsWith('file:') || isGitHostedPkgUrl((pkgSnapshot.resolution as TarballResolution).tarball ?? '')
```

Comment:
This change doesn't really make a difference because git-hosted deps should not have a registry field.